scene/2d/navigation_region_2d.cpp

```diff
@@ -553,7 +558,7 @@ void NavigationRegion2D::_navpoly_changed() {
 	}
 }
 void NavigationRegion2D::_map_changed(RID p_map) {
-	if (enabled && get_world_2d()->get_navigation_map() == p_map) {
+	if (is_inside_tree() && get_world_2d()->get_navigation_map() == p_map) {
 		update();
 	}
```

Comment:
Should this be enclosed in `#ifdef DEBUG_ENABLED` ? I.e. if you are not going to use a draw, you shouldn't be triggering an update I think?

Reply:
Can be, the signal is already enclosed so wouldnt fire without a debug build anyway.